src/wallet/coinselection.cpp

```diff
@@ -89,7 +89,7 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
     CAmount best_waste = MAX_MONEY;
 
     // Depth First search loop for choosing the UTXOs
-    for (size_t curr_try = 0, curr_index = 0; curr_try < TOTAL_TRIES; ++curr_try, ++curr_index) {
+    for (size_t curr_try = 0, utxo_pool_index = 0; curr_try < TOTAL_TRIES; ++curr_try, ++utxo_pool_index) {
```

Comment:
33e8d4b4b8999d5f80227fa20bed7271ab0976bd should be squashed into cec31e08ca64168fd084bc71ec250d8eaca99537, it was confusing to me that we're introducing a new `curr_index` variable in one commit, and then renaming it 2 commits later